src/components/elements/input/index.tsx

```diff
@@ -0,0 +1,18 @@
+import { StyledInput, StyledInputBorder, StyledInputWrapper } from './input.styles';
+import type { InputProps } from './input.types';
+import { Text } from '../text';
+
+export const Input = ({ as = 'input', errorMessage, hasError = false, ...props }: InputProps) => {
```

Comment:
Can we use forwardRef here? This way, we will be able to access the input's properties and methods, such as focus, select, or blur.